### PR TITLE
fix: using custom web components in form fields

### LINKED ISSUE #410: Example in  "Using a Field Web Component" is not working
The code in this page is not working:
https://vaadin.com/docs/latest/fusion/forms/appendix-web-component-field-strategy

In particular:

```typescript
import {AbstractModel, Binder, BinderConfiguration, FieldStrategy, ModelConstructor} from '@vaadin/form';
import { MyCcStrategy } from './my-cc-strategy';

export class MyBinder<T, M extends AbstractModel<T>> extends Binder<T, M> {
    constructor(context: any, model: ModelConstructor<T, M>, config?: BinderConfiguration<T>) {
        super(context, model, config);
    }

    getFieldStrategy(element: any): FieldStrategy {
        if (element.localName === 'my-cc-field') {
            return new MyCcStrategy(element);
        }
        return super.getFieldStrategy(element);
    }
}
```

And:
`onValueChanged(event) {` instead of `onValueChanged(event:CustomFieldValueChanged) {`

To avoid this, it would be better to create a separate example in Typescript that runs and embed this example.

### PR DESCRIPTION
**Pull request by vlukashov**
- code snippets are updated so that they actually compile and work
 - the text content is rephrased to clarify the purpose and the general principles behind using custom web components with the client-side form binder API.

Fixes #410

### CODE REVIEW

**Comment by jcgueriaud1** [comment]
I think this PR https://github.com/vaadin/docs/pull/418 fixes the same issue: https://github.com/vaadin/docs/issues/410

I think only one should be merged and the other closed. (Mine doesn't change the text, only fixes and externalize the examples)

File: articles/fusion/forms/appendix-web-component-field-strategy.asciidoc [hunk 1]
```diff
@@ -1,125 +1,58 @@
 ---
-title: Using a Field Web Component
+title: Binding Data to Custom Components
 order: 100
 layout: page
 ---
 
 
-= Binding a Field Web Component
+= Using Custom Web Components in a Form
```
Comment (haijian-vaadin) [inline articles/fusion/forms/appendix-web-component-field-strategy.asciidoc:8]: Maybe it's better to also update the title here to `Binding Data to Custom Components` keep it in sync with the menu item?

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]

**Comment by vlukashov** [comment]
> I think this PR #418 fixes the same issue: #410
> 
> I think only one should be merged and the other closed. (Mine doesn't change the text, only fixes and externalize the examples)

I should have checked the open issues and PRs before starting to fix this on my own :(
Now both me and @jcgueriaud1 have done the same work twice.

I would probably take the PR with the changes to the text content. Would you like to suggest changes to the code samples in this PR, @jcgueriaud1?

File: articles/fusion/forms/appendix-web-component-field-strategy.asciidoc (hunk 1 above)
Comment (vlukashov) [reply articles/fusion/forms/appendix-web-component-field-strategy.asciidoc]: 👍

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**